src/main/java/org/apache/flink/graph/streaming/library/ConnectedComponents.java

```diff
@@ -0,0 +1,129 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.graph.streaming.library;
+
+import org.apache.flink.api.common.functions.GroupReduceFunction;
+import org.apache.flink.api.common.functions.ReduceFunction;
+import org.apache.flink.graph.streaming.EdgesFold;
+import org.apache.flink.graph.streaming.GraphWindowStream;
+import org.apache.flink.graph.streaming.WindowGraphAggregation;
+import org.apache.flink.graph.streaming.example.IterativeConnectedComponents;
+import org.apache.flink.graph.streaming.example.util.DisjointSet;
+import org.apache.flink.types.NullValue;
+
+import java.io.Serializable;
+
+/**
+ * The Connected Components algorithm assigns a component ID to each vertex in the graph.
+ * Vertices that belong to the same component have the same component ID.
+ * This algorithm computes _weakly_ connected components, i.e. edge direction is ignored.
+ * <p>
+ * This is a single-pass implementation, which uses a {@link WindowGraphAggregation} to periodically merge
+ * the partitioned state. For an iterative implementation, see {@link IterativeConnectedComponents}.
+ *
+ * @param <K>  the vertex ID type
+ * @param <EV> the edge value type
+ */
+public class ConnectedComponents<K extends Serializable, EV> extends WindowGraphAggregation<K, EV, DisjointSet<K>, DisjointSet<K>> implements Serializable {
+
+	private long mergeWindowTime;
+
+	public ConnectedComponents(long mergeWindowTime) {
+		super(new UpdateCC(), new CombineCC(), new DisjointSet<K>(), mergeWindowTime, false);
+	}
+
+	/**
+	 * Implements EdgesFold Interface, applies a function to a vertex neighborhood
+	 * in the {@link GraphWindowStream#foldNeighbors(Object, EdgesFold)} method.
+	 *
+	 * @param <K> the vertex ID type
+	 */
+	public final static class UpdateCC<K extends Serializable> implements EdgesFold<K, NullValue, DisjointSet<K>> {
+
+		/**
+		 * Combines two edge values into one value of the same type.
+		 * The foldEdges function is consecutively applied to all edges of a neighborhood,
+		 * until only a single value remains.
+		 *
+		 * @param ds        the initial value and accumulator
+		 * @param vertex    the vertex ID
+		 * @param vertex2   the neighbor's ID
+		 * @param edgeValue the edge value
+		 * @return The data stream that is the result of applying the foldEdges function to the graph window.
+		 * @throws Exception
+		 */
```

Comment:
Same as the previous comments. Copying the javadoc isn't helpful. You should describe what this implementation is doing instead :)